kbase-extension/static/kbase/js/util/jobLogViewer.js

```diff
@@ -956,29 +1117,34 @@ define([
                 },
                 handle: handleJobLogs,
             });
-            listeners.push(ev);
 
             // An error may encountered during the job fetch on the back end. It is
             // reported as a job-comm-error, but translated by the jobs panel as a job-log-deleted
             // TODO: we may want to rethink this. It might be beneficial to emit this as the
             // original error, and let the widget sort it out? Or perhaps on the back end
             // have the error emitted as a specific message. It is otherwise a little difficult
             // to trace this.
-            ev = runtime.bus().listen({
+            listenersByType['job-log-deleted'] = bus.listen({
                 channel: {
                     jobId: jobId,
                 },
                 key: {
                     type: 'job-log-deleted',
                 },
-                handle: function () {
-                    stopAutoFetch();
-                    render();
+                handle: function (message) {
+                    awaitingLog = false;
+                    stopLogAutoFetch();
+                    renderLog();
+                    if (
+                        message.message &&
+                        message.message.indexOf('Cannot find job log with id') === 0
+                    ) {
+                        fsm.newState({ mode: 'job-not-found' });
+                    }
```

Comment:
Are there other reasons why a job log would be deleted, apart from just that the job ID is not found?